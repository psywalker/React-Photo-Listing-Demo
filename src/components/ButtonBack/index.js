import React from 'react';
import { Button, Icon } from 'antd';
import { withRouter } from 'react-router-dom';
import { withLastLocation } from 'react-router-last-location';
import './index.scss';

export const ButtonBack = withRouter(({ history, lastLocation }) => {
  if (!lastLocation) return false;
  return (
    <Button
      data-test="btnBack"
      className="btn-back"
      type="default"
      onClick={() => {
        history.push(lastLocation.pathname);
      }}
    >
      <Icon type="left" />
      Go back
    </Button>
  );
});

export default withLastLocation(ButtonBack);
