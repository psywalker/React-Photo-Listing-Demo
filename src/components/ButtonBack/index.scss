.btn-back {
  margin-left: 10px;
  height: 37px !important;
  line-height: 35px !important;
  top: -6px;
	@media screen and ( min-width: 0px ) and ( max-width: 400px ) {
    padding: 0 5px !important;  
    margin-left: 6px;
    & span {
      margin-left: 0px !important;
    }
  }
}