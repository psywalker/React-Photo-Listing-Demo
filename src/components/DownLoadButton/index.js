import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { saveAs } from 'file-saver';
import {
  Button,
  Icon,
} from 'antd';
import handleDowloadPhoto from '../../utils/handleDowloadPhoto';
import './index.scss';

const DownLoadButton = memo(({
  altDescriprion,
  photoDesc,
  photoSrc,
  textButton = null,
}) => {
  const downLoadPhoto = () => {
    const photoName = handleDowloadPhoto(altDescriprion, photoDesc);
    saveAs(photoSrc, photoName);
  };
  return (
    <div
      data-test="downloadButtonContainer"
      className="download-button"
    >
      <Button
        data-test="downloadButton"
        className="download-button__btn"
        onClick={downLoadPhoto}
      >
        <Icon
          data-test="downloadButtonIcon"
          className="download-button__icon"
          type="download"
        />
        { textButton && <div className="download-button__text">{ textButton }</div> }
      </Button>
    </div>
  );
});

DownLoadButton.propTypes = {
  altDescriprion: PropTypes.string,
  photoDesc: PropTypes.string,
  photoSrc: PropTypes.string,
  textButton: PropTypes.string,
};
DownLoadButton.defaultProps = {
  altDescriprion: '',
  photoDesc: '',
  photoSrc: '',
  textButton: '',
};

export default DownLoadButton;
