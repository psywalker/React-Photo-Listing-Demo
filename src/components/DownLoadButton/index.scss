.download-button {
  &__popover {
    
  }
  &__btn.ant-btn {
    padding: 0 10px;

    & span {
      margin-left: 0;
    }
  }
  .ant-btn > .anticon + span, .ant-btn > span + .anticon {
    margin-left: 0px;
  }
  &__text {
    margin-left: 5px;
    display: inline;
  }
}
