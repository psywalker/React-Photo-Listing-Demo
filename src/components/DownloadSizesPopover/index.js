import React, { memo, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { saveAs } from 'file-saver';
import { useTranslation } from 'react-i18next';
import {
  Button,
  Divider,
  Popover,
} from 'antd';
import handleDowloadPhoto from '../../utils/handleDowloadPhoto';
import handleVisibleByScroll from '../../utils/handleVisibleByScroll';
import './index.scss';

const DownloadSizesPopover = memo(({
  photoUrlSizes,
  userID,
  title,
  altDescriprion,
  photoDesc,
  placement,
  children,
  handleVisibleButtonChangeFalse,
}) => {
  const [popupVisible, setPopupVisible] = useState(false);
  const { t } = useTranslation();
  const downLoadPhoto = (url) => {
    const photoName = handleDowloadPhoto(altDescriprion, photoDesc);
    saveAs(url, photoName);
  };

  const handleVisibleChange = (visible) => {
    setPopupVisible(visible);
    if (!visible) handleVisibleButtonChangeFalse();
  };
  const hidePopover = () => {
    setPopupVisible(false);
  };

  useEffect(() => {
    handleVisibleByScroll('addEventListener', ['scroll', 'resize'], [hidePopover, handleVisibleButtonChangeFalse]);
    return () => {
      handleVisibleByScroll('removeEventListener', ['scroll', 'resize'], [hidePopover, handleVisibleButtonChangeFalse]);
    };
  });

  return (
    <Popover
      overlayClassName="download-button__popover download-button-popover"
      data-test="photoCardPopover"
      placement={placement}
      title={t('changePhotoSize')}
      visible={popupVisible}
      onVisibleChange={handleVisibleChange}
      content={(
        <div className="download-button__popup download-sizes">
          { photoUrlSizes.map(item => (
            <Button
              key={item.value}
              className="download-sizes__btn"
              size="small"
              onClick={() => {
                downLoadPhoto(item.url);
                hidePopover();
                handleVisibleButtonChangeFalse();
              }}
            >
              {t(item.value)}
            </Button>
          ))}
          <Divider style={{ margin: '5px 0' }} />
          <div className="download-sizes__footer credit">
            {t('photoBy')}
            {' '}
            <a
              href={`https://unsplash.com/@${userID}?utm_medium=referral&amp;utm_campaign=photographer-credit&amp;utm_content=creditBadge`}
              target="_blank"
              rel="noopener noreferrer"
            >
              {title}
            </a>
            {' '}
            {t('photoOn')}
            {' '}
            <a
              href="https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
              target="_blank"
              rel="noopener noreferrer"
            >
              Unsplash
            </a>
          </div>
        </div>

      )}
      trigger="click"
    >
      { children }
    </Popover>
  );
});

DownloadSizesPopover.propTypes = {
  userID: PropTypes.string,
  title: PropTypes.string,
  altDescriprion: PropTypes.string,
  photoDesc: PropTypes.string,
  placement: PropTypes.string,
  photoUrlSizes: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string,
    value: PropTypes.string,
  })),
  handleVisibleButtonChangeFalse: PropTypes.func,
  children: PropTypes.shape({}),
};

DownloadSizesPopover.defaultProps = {
  userID: '',
  title: '',
  altDescriprion: '',
  photoDesc: '',
  placement: '',
  photoUrlSizes: [],
  handleVisibleButtonChangeFalse: () => {},
  children: {},
};

export default DownloadSizesPopover;
