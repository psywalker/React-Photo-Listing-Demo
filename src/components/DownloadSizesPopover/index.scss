.download-sizes {
  margin-left: -5px;
  &__btn.ant-btn {
    margin-left: 5px;
    margin-bottom: 5px;
  }
}
.credit {
  padding-left: 5px;
}

.download-button-popover.ant-popover {
  .ant-popover-content > .ant-popover-arrow {
    display: none;
  }
}