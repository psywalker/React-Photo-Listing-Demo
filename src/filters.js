const filters = [
  {
    id: 0,
    label: 'Editorial',
    filterValue: 'editorial',
  },
  {
    id: 1,
    label: 'Following',
    filterValue: 'following',
    border: true,
  },
  {
    id: 2,
    label: 'Wallpapers',
    filterValue: 'wallpapers',
  },
  {
    id: 3,
    label: 'Textures & Patterns',
    filterValue: 'textures',
  },
  {
    id: 4,
    label: 'Nature',
    filterValue: 'nature',
  },
  {
    id: 5,
    label: 'Current Events',
    filterValue: 'events',
  },
  {
    id: 6,
    label: 'Architecture',
    filterValue: 'architecture',
  },
  {
    id: 7,
    label: 'Business & Work',
    filterValue: 'business',
  },
  {
    id: 8,
    label: 'Animals',
    filterValue: 'animals',
  },
  {
    id: 9,
    label: 'Travel',
    filterValue: 'travel',
  },
  {
    id: 10,
    label: 'Fashion',
    filterValue: 'fashion',
  },
  {
    id: 11,
    label: 'Food & Drink',
    filterValue: 'food',
  },
  {
    id: 12,
    label: 'Spirituality',
    filterValue: 'spirituality',
  },
  {
    id: 13,
    label: 'Experimental',
    filterValue: 'experimental',
  },
  {
    id: 14,
    label: 'People',
    filterValue: 'people',
  },
  // {
  //   id: 15,
  //   label: 'Health',
  //   filterValue: 'health',
  // },
  // {
  //   id: 16,
  //   label: 'Arts & Culture',
  //   filterValue: 'arts',
  // },
];
export default filters;
