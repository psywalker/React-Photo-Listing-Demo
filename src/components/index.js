import ButtonBack from './ButtonBack';
import HeaderApp from './HeaderApp';
import NavTop from './NavTop';
import PhotoCard from './PhotoCard';
import Search from './Search';
import Spinner from './Spinner';
import SpinnerPhoto from './SpinnerPhoto';
import InfoPhotoModal from './InfoPhotoModal';
import Error from './Error';
// import HighchartsHOC from './HighchartsHOC';
import SmallPhotoListing from './SmallPhotoListing';
import Tags from './Tags';
import HomeTestHOC from '../pages/Home/HomeTestHOC';
import Photo from '../pages/Photo';
import Profile from '../pages/Profile';
import User from '../pages/User';
import UserStatistic from '../pages/UserStatistic';

export {
  ButtonBack,
  HeaderApp,
  NavTop,
  PhotoCard,
  Search,
  Spinner,
  SpinnerPhoto,
  InfoPhotoModal,
  Error,
  HomeTestHOC,
  Photo,
  Profile,
  User,
  UserStatistic,
  // HighchartsHOC,
  SmallPhotoListing,
  Tags,
};
