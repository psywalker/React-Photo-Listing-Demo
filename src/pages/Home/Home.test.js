import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';
import { shape } from 'prop-types';
import { createSerializer } from 'enzyme-to-json';
import { Home } from '.';
import { Search, Photo } from '../../components';
import filters from '../../filters';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
Enzyme.configure({ adapter: new Adapter() });

describe('Test of component of Home', () => {
  const initialProps = {
    cardsPhotosRequestAction: () => {},
    paginationChangeAction: () => {},
    filterItemValueAction: () => {},
    searchTextAction: () => {},
    searchChangeInputValueAction: () => {},
    filters: [],
    isListingLoading: false,
    cards: [],
    cardsData: {
      query: 'wallpapers',
      page: 1,
      per_page: 6,
    },
    totalCards: 0,
    navTopItemActive: 2,
    photolistingRequestError: false,
    match: {
      params: {
        tag: {},
      },
    },
  };
  const router = {
    history: new BrowserRouter().history,
    route: {
      location: {},
      match: {},
    },
  };

  const createContext = () => ({
    context: { router },
    childContextTypes: { router: shape({}) },
  });

  function mountWrap(node) {
    return mount(node, createContext());
  }

  describe('Home component initial', () => {
    it('renders with initial props', () => {
      const home = shallow(<Home {...initialProps} />);

      expect(home.find('[data-test="navTopItem"]')).toHaveLength(0);
      expect(home.find('[data-test="CardsTextEmpty"]')).toHaveLength(1);
      expect(home.find('[data-test="pagination"]')).toHaveLength(1);
      expect(home.find('[data-test="spinner"]')).toHaveLength(0);
      expect(home).toMatchSnapshot();
    });

    it('renders with loading', () => {
      const props = {
        ...initialProps,
        isListingLoading: true,
      };

      const home = shallow(<Home {...props} />);
      expect(home.find('[data-test="spinner"]')).toHaveLength(1);
      expect(home).toMatchSnapshot();
    });

    it('renders without loading', () => {
      const props = {
        ...initialProps,
        isListingLoading: false,
      };

      const home = shallow(<Home {...props} />);
      expect(home.find('[data-test="spinner"]')).toHaveLength(0);
      expect(home).toMatchSnapshot();
    });

    it('renders with error', () => {
      const props = {
        ...initialProps,
        photolistingRequestError: true,
      };

      const home = shallow(<Home {...props} />);
      expect(home.find('[data-test="errorText"]').text()).toEqual('Error loading photolisting');
      expect(home).toMatchSnapshot();
    });

    it('renders Error if `totalCards` > `cardsData.per_page`', () => {
      const props = {
        ...initialProps,
        totalCards: 7,
        cardsData: {
          query: 'wallpapers',
          page: 1,
          per_page: 2,
        },
      };

      const home = mountWrap(<Home {...props} />);
      expect(home.find('[data-test-id-pagination="pagination"]')).toHaveLength(2);
      expect(home).toMatchSnapshot();
    });

    it('dispatches the `cardsPhotosRequestAction()` method it receives from props', () => {
      const mockFetchRequestAction = jest.fn();
      const props = {
        ...initialProps,
        cardsPhotosRequestAction: mockFetchRequestAction,
        cardsData: {
          query: 'wallpapers',
          page: 1,
          per_page: 2,
        },
      };

      const home = shallow(<Home {...props} />);
      home.setProps({
        cardsData: {
          query: 'wallpapers',
          page: 1,
          per_page: 2,
        },
      });
      expect(mockFetchRequestAction).toHaveBeenCalledTimes(1);
      expect(home).toMatchSnapshot();
    });

    it('dispatches the `getPaginationChange()` method it receives from props', () => {
      const mockFetchRequestAction = jest.fn();
      const props = {
        ...initialProps,
        paginationChangeAction: mockFetchRequestAction,
      };

      const home = shallow(<Home {...props} />);
      home.setProps({
        cardsData: {
          query: 'wallpapers',
          page: 1,
          per_page: 2,
        },
      });
      home.instance().getPaginationChange();
      expect(mockFetchRequestAction).toHaveBeenCalledTimes(1);
      expect(home).toMatchSnapshot();
    });
  });

  describe('Home component with content', () => {
    it('renders without filters for NavTop', () => {
      const props = {
        ...initialProps,
      };

      const home = shallow(<Home {...props} />);
      expect(home.find('.nav-top__item')).toHaveLength(0);
      expect(home).toMatchSnapshot();
    });
  });

  describe('Mount tests', () => {
    it('Test method of tags', () => {
      const { id, filterValue } = filters[0];
      const props = {
        ...initialProps,
        filters,
      };

      const home = mountWrap(<Home {...props} />);
      home.instance().getFilterItemValue = jest.fn();
      home.instance().forceUpdate();
      home.find('.ant-tag').first().simulate('click');
      expect(home.instance().getFilterItemValue).toHaveBeenCalledTimes(1);
      expect(home.instance().getFilterItemValue).toHaveBeenCalledWith(filterValue, id);
    });
  });
});