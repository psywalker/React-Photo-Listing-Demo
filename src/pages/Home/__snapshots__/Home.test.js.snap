// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component of Home Home component initial dispatches the \`cardsPhotosRequestAction()\` method it receives from props 1`] = `
<div
  className="App"
>
  <Search
    onChangeInputValue={[Function]}
    onSearchInputValue={[Function]}
    queryText="wallpapers"
  />
  <ul
    className="nav-top"
  />
  <ul
    className="photo-list"
  />
  <div
    className="cards__text-empty"
  >
    <Memo()
      smallErrorFlag={false}
      text="No images were found for your request. Try to find more."
    />
  </div>
  <div
    className="pagination"
  />
</div>
`;

exports[`Test of component of Home Home component initial dispatches the \`getPaginationChange()\` method it receives from props 1`] = `
<div
  className="App"
>
  <Search
    onChangeInputValue={[Function]}
    onSearchInputValue={[Function]}
    queryText="wallpapers"
  />
  <ul
    className="nav-top"
  />
  <ul
    className="photo-list"
  />
  <div
    className="cards__text-empty"
  >
    <Memo()
      smallErrorFlag={false}
      text="No images were found for your request. Try to find more."
    />
  </div>
  <div
    className="pagination"
  />
</div>
`;

exports[`Test of component of Home Home component initial renders with error 1`] = `
<div
  className="error-text"
>
  Error loading photolisting
</div>
`;

exports[`Test of component of Home Home component initial renders with initial props 1`] = `
<div
  className="App"
>
  <Search
    onChangeInputValue={[Function]}
    onSearchInputValue={[Function]}
    queryText="wallpapers"
  />
  <ul
    className="nav-top"
  />
  <ul
    className="photo-list"
  />
  <div
    className="cards__text-empty"
  >
    <Memo()
      smallErrorFlag={false}
      text="No images were found for your request. Try to find more."
    />
  </div>
  <div
    className="pagination"
  />
</div>
`;

exports[`Test of component of Home Home component initial renders with loading 1`] = `
<div
  className="App"
>
  <Memo()
    className="spinner"
  />
  <Search
    onChangeInputValue={[Function]}
    onSearchInputValue={[Function]}
    queryText="wallpapers"
  />
  <ul
    className="nav-top"
  />
  <div
    className="cards__text-empty"
  >
    <Memo()
      smallErrorFlag={false}
      text="No images were found for your request. Try to find more."
    />
  </div>
  <div
    className="pagination"
  />
</div>
`;

exports[`Test of component of Home Home component with content renders with cards 1`] = `
<div
  className="App"
>
  <Search
    onChangeInputValue={[Function]}
    onSearchInputValue={[Function]}
    queryText="wallpapers"
  />
  <ul
    className="nav-top"
  />
  <ul
    className="photo-list"
  >
    <li
      className="photo-list__item"
      key="kdGstD3te3M"
    >
      <Memo()
        onSearchTagValue={[Function]}
        photoDesc={null}
        photoID="kdGstD3te3M"
        photoName="https://images.unsplash.com/photo-1558981408-db0ecd8a1ee4?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjQzODA4fQ"
        tags={Array []}
        title="Harley-Davidson"
        userAvatar="https://images.unsplash.com/profile-1556751276456-1561737ea797?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128"
        userID="harleydavidson"
      />
    </li>
  </ul>
  <div
    className="cards__text-empty"
  >
    <Memo()
      smallErrorFlag={false}
      text="No images were found for your request. Try to find more."
    />
  </div>
  <div
    className="pagination"
  />
</div>
`;

exports[`Test of component of Home Mount tests renders with cards 1`] = `
<div
  className="App"
>
  <div
    className="search"
  >
    <span
      className="ant-input-search search__input ant-input-affix-wrapper"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <input
        className="ant-input"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Wallpapers"
        style={null}
        type="text"
        value="wallpapers"
      />
      <span
        className="ant-input-suffix"
      >
        <i
          aria-label="icon: search"
          className="anticon anticon-search ant-input-search-icon"
          onClick={[Function]}
          tabIndex={-1}
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="search"
            fill="currentColor"
            focusable="false"
            height="1em"
            key="svg-search"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              key="svg-search-svg-0"
            />
          </svg>
        </i>
      </span>
    </span>
  </div>
  <ul
    className="nav-top"
  />
  <ul
    className="photo-list"
  >
    <li
      className="photo-list__item"
      key="kdGstD3te3M"
    >
      <div
        className="photo-card-self"
      >
        <div
          className="ant-card ant-card-bordered"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="ant-card-cover"
          >
            <a
              className="photo-card__photo-link"
              href="/photo/kdGstD3te3M"
              onClick={[Function]}
            >
              <img
                alt="example"
                className="photo-card__img"
                src="https://images.unsplash.com/photo-1558981408-db0ecd8a1ee4?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjQzODA4fQ"
              />
            </a>
          </div>
          <div
            className="ant-card-body"
            style={Object {}}
          >
            <a
              className="photo-card-self__link-ava"
              href="/users/harleydavidson"
              onClick={[Function]}
            >
              <div
                className="ant-card-meta"
              >
                <div
                  className="ant-card-meta-avatar"
                >
                  <span
                    className="ant-avatar ant-avatar-circle ant-avatar-image"
                    style={Object {}}
                  >
                    <img
                      onError={[Function]}
                      src="https://images.unsplash.com/profile-1556751276456-1561737ea797?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128"
                    />
                  </span>
                </div>
                <div
                  className="ant-card-meta-detail"
                >
                  <div
                    className="ant-card-meta-title"
                  >
                    Harley-Davidson
                  </div>
                </div>
              </div>
            </a>
            <p
              className="photo-card-self__desc"
            >
              No Description
            </p>
            <div
              className="photo-card-self__badge-wrap"
            />
          </div>
        </div>
      </div>
    </li>
  </ul>
  <div
    className="cards__text-empty"
  >
    <div
      className="error "
    >
      <div
        className="error__content"
      >
        <div
          className="error__content-inner"
        >
          <h3
            className="error__title"
          >
            OOPS
          </h3>
          <h2
            className="error__subtitle"
          >
            No images were found for your request. Try to find more.
          </h2>
           
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="ant-btn"
              onClick={[Function]}
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
              type="button"
            >
              <span
                key=".0"
              >
                BACK TO HOME
              </span>
            </button>
          </a>
        </div>
      </div>
      <img
        alt="No images were found for your request. Try to find more."
        className="error__bg"
        src="http://localhost:3000/error-image.jpg"
      />
    </div>
  </div>
  <div
    className="pagination"
  />
</div>
`;
